src/decommit.rs

```diff
@@ -77,6 +77,17 @@ impl WorldDiff {
         Some((UnpaidDecommit { cost, code_key }, is_evm))
     }
 
+    pub(crate) fn decommit_opcode(
+        &mut self,
+        world: &mut dyn World,
+        code_hash: U256,
+    ) -> Option<Program> {
```

Comment:
This should not return a Program, just the code page. You also need to change the World interface because we do not want to decode Programs when we actually only need the code page.